letsencrypt_nginx/configurator.py

```diff
@@ -274,17 +291,22 @@ def _make_server_ssl(self, vhost):
         :type vhost: :class:`~letsencrypt_nginx.obj.VirtualHost`
 
         """
-        ssl_block = [['listen', '443 ssl'],
-                     ['ssl_certificate',
-                      '/etc/ssl/certs/ssl-cert-snakeoil.pem'],
-                     ['ssl_certificate_key',
-                      '/etc/ssl/private/ssl-cert-snakeoil.key'],
+        snakeoil_cert, snakeoil_key = self._get_snakeoil_paths()
+        ssl_block = [['listen', '{0} ssl'.format(self.config.dvsni_port)],
```

Comment:
I think this goes into the general category of finding the correct virtualhost.  I recently added a Vhost selection screen for Apache. https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt_apache/display_ops.py that allows the user to choose a vhost when there isn't a clear selection.  If it is of any use, I will move it into a common file within the core.

Note: that the vhost selection does not solve the problem of when you have to upgrade a preexisting vhost.  That would still require the ```--ssl-vhost-port``` option